From: Mike Marciniszyn <mike.marciniszyn@intel.com>
Date: Thu, 12 May 2016 10:22:57 -0700
Subject: [PATCH] IB/hfi1: Fix pio wait counter double increment
Patch-mainline: v4.7-rc1
Git-commit: 17f15bf66884b9e33c5005ee0149b039af8f7af2
References: bsc#973818 FATE#319242

The code unconditionlly increments the pio wait counter
making the counter inacurate and unusable.

Fixes: 14553ca11039 ("staging/rdma/hfi1: Adaptive PIO for short messages")
Reviewed-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Signed-off-by: Mike Marciniszyn <mike.marciniszyn@intel.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/staging/rdma/hfi1/verbs.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/staging/rdma/hfi1/verbs.c
+++ b/drivers/staging/rdma/hfi1/verbs.c
@@ -946,7 +946,6 @@ static int pio_wait(struct rvt_qp *qp,
 
 			dev->n_piowait += !!(flag & RVT_S_WAIT_PIO);
 			dev->n_piodrain += !!(flag & RVT_S_WAIT_PIO_DRAIN);
-			dev->n_piowait++;
 			qp->s_flags |= flag;
 			was_empty = list_empty(&sc->piowait);
 			list_add_tail(&priv->s_iowait.list, &sc->piowait);
